### ISSUE TITLE
LogRecord->with does not transfers "formatted" value

### ISSUE THREAD

**Issue by Justinas-Jurciukonis** [Seldaek/monolog#1788]
Monolog version 3.2.0

When `PsrLogMessageProcessor` processes message, it returns new instance of `LogRecord` using code

```php
return $record->with(message: strtr($record->message, $replacements), context: $context);
```

and inside [`LogRecord->with`](https://github.com/Seldaek/monolog/blob/main/src/Monolog/LogRecord.php#L118) there is missing `formatted` key to be copied over, so if any previous processor set any information inside `LogRecord->formatted` it will be lost.

Seems like same issue with `toArray`

**Comment by Seldaek** [comment]
Normally formatting happens after processing, so this should never be a problem. Have you experienced an actual issue?

**Comment by stof** [comment]
This looks like the use case for `LogRecord->extra` instead

**Comment by Seldaek** [comment]
Right, exactly my thinking @stof - sounds to me like these processors should all record stuff in extra as they are meant to, and then the final formatter can use the info in context/extra to generate the final formatted record.
